### PR TITLE
Pro 4257 modal operation

### PR DESCRIPTION
**Pull request by falkodev**
## Summary

Add the ability to have custom modals for batch operations, and align props between doc context menu custom modal and batch operation custom modal, in order to use the same component.

## What are the specific steps to test this change?

Test with https://github.com/apostrophecms/import-export/pull/2

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [x] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [x] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [x] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

### CODE REVIEW

File: modules/@apostrophecms/doc-type/ui/apos/components/AposDocContextMenu.vue [hunk 1]
```diff
@@ -428,13 +428,18 @@ export default {
 
     },
     async customAction(doc, operation) {
+      if (operation.label) {
```
Comment (boutell) [inline modules/@apostrophecms/doc-type/ui/apos/components/AposDocContextMenu.vue:431]: How is this related to the label? 🤔

Comment (boutell) [reply modules/@apostrophecms/doc-type/ui/apos/components/AposDocContextMenu.vue:431]: Is the idea that label and modal are mutually exclusive? Maybe better to check that `operation.moduleName` is not set before doing this.

**Review by boutell — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/doc-type/ui/apos/components/AposDocContextMenu.vue (hunk 1 above)
Comment (falkodev) [reply modules/@apostrophecms/doc-type/ui/apos/components/AposDocContextMenu.vue:431]: `operation.moduleName` is set with the wrong module name and is used in some modal, so I guess there is a use case where it's valid. In my use case, I need the current module name.

I've change this `if` condition with a custom property I've created on the backend instead. Will it be safe @boutell ?

File: modules/@apostrophecms/modal/ui/apos/components/AposDocsManagerToolbar.vue [hunk 2]
```diff
@@ -263,7 +263,18 @@ export default {
...
 
-      await this.confirmOperation(operation);
+      operation.modal ? await this.modalOperation(operation) : await this.confirmOperation(operation);
+    },
+    async modalOperation({
+      modal, ...rest
+    }) {
+      await apos.modal.execute(modal, {
+        count: this.checkedCount,
+        moduleName: this.checkedCount === 1
```
Comment (boutell) [inline modules/@apostrophecms/modal/ui/apos/components/AposDocsManagerToolbar.vue:273]: Wait, why would the label be the source of the Apostrophe module name? I'm a little lost now.

**Review by boutell — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue [hunk 3]
```diff
@@ -72,6 +72,7 @@
             :is-relationship="!!relationshipField"
             :checked-count="checked.length"
             :batch-operations="moduleOptions.batchOperations"
+            :module-name="moduleName"
```
Comment (boutell) [inline modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue:75]: AposMediaManager should pass it too. And the doc template library one, in a separate PR.

**Review by boutell — CHANGES_REQUESTED** [review]
